users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -106,13 +116,90 @@ def add_specaug_layer(
     # Update config with pre-generated parameters
     full_config["specaug_params"] = specaug_params
 
-    network[name] = {
-        "class": "eval",
-        "from": from_list,
-        "eval": f"self.network.get_config().typed_value('transform')(source(0, as_data=True), network=self.network, **{full_config})",
-    }
+    if not full_config["mask_based_on_variance"]:
+
+        network[name] = {
+            "class": "eval",
+            "from": from_list,
+            "eval": f"self.network.get_config().typed_value('transform')(source(0, as_data=True), network=self.network, **{full_config})",
+        }
+        return [name], [
+            sort_filters_by_center_freq,
+            get_frequency_response,
+            _mask,
+            random_mask,
+        ]
+    else:
+        network[name] = {
+            "class": "eval",
+            "from": from_list,
+            "eval": f"self.network.get_config().typed_value('transform_with_variance')(source(0, as_data=True), network=self.network, **{full_config})",
+        }
+        return [name], [
+            get_frequency_response,
+            calculate_frequency_response_variance,
+            _mask,
+            random_mask,
+            variance_based_mask,
+            transform_with_variance,
+        ]
+
+
+def calculate_frequency_response_variance(filter_layer):
+    """
+    Calculate the variance of the frequency response for each feature.
+
+    :param tf.Tensor filter_layer: The filter layer tensor
+    :return: Variance of frequency response for each feature
+    :rtype: tf.Tensor
+    """
+    import tensorflow as tf
+
+    f_resp = get_frequency_response(filter_layer)
+
+    # Calculate variance for each feature
+    variance = tf.math.reduce_variance(f_resp, axis=0)
+    return variance
+
+
+def variance_based_mask(x, batch_axis, axis, variance, variance_factor, max_number_masks_for_variance_specaug):
```

Comment:
In light of my previous general comment on generality: Maybe it would be better to have the probabilities per channel as input here and compute them in the transform function. Then you could just replace the transform function if you want to test another way of computing the probabilities.